src/Entity.php

```diff
@@ -67,34 +67,51 @@ protected function request($method, $relativeUrl, $data = null)
         }
         else
         {
-            return static::buildEntity($response);
+            return static::buildEntity(null, $response);
         }
     }
 
-    protected static function buildEntity($data)
+    protected static function buildEntity($key, $data)
     {
         $entities = static::getDefinedEntitiesArray();
 
-        if (isset($data['entity']))
+        if (self::isNotes($key) === true)
```

Comment:
make a generic isArrayableAttribute function.
Shift 'notes' to an array in the class.